Sources/StreamChatUI/ChatChannelList/ChatChannelListItemView.swift

```diff
@@ -188,6 +217,19 @@ open class ChatChannelListItemView: _View, ThemeProvider, SwiftUIRepresentable {
 
         unreadCountView.content = content?.channel.unreadCount ?? .noUnread
         unreadCountView.invalidateIntrinsicContentSize()
+        
+        let checkmarkContent = previewMessageDeliveryStatus.map {
+            ChatMessageDeliveryStatusCheckmarkView.Content(deliveryStatus: $0)
+        }
+        previewMessageDeliveryStatusView.content = checkmarkContent
+        previewMessageDeliveryStatusView.isHidden = checkmarkContent == nil
+        
+        if let status = checkmarkContent?.deliveryStatus {
+            bottomContainer.insertArrangedSubview(
+                previewMessageDeliveryStatusView,
+                at: status == .pending || status == .failed ? 0 : 1
+            )
+        }
```

Comment:
Nit:
```suggestion
        if let status = previewMessageDeliveryStatus {
            bottomContainer.insertArrangedSubview(
                previewMessageDeliveryStatusView,
                at: status == .pending || status == .failed ? 0 : 1
            )
        }
```

Reply:
`previewMessageDeliveryStatus` is a computed property so accessing it one more time would run the calculation again.

Reply:
It's not like we are doing an intensive task here 😛 I would prefer readability over premature optimization, the calculation here is 0.0000001 sec

Reply:
Readability is fine in both cases, I think. Running the code more than it's needed is a bad practice and can bite us back if the computation becomes heavier (e.g. `readByCount` will be changed to always make a count request to DB). I'd prefer to keep the variable in place 👍